File: examples-sdk/swift/010-basic-websocket-chat/main.swift
// 010-basic-websocket-chat — Swift SDK equivalent
// Compile: vil compile --from swift --input 010-basic-websocket-chat/main.swift --release

let server = VilServer(name: "websocket-chat", port: 8080)
let chat = ServiceProcess(name: "chat")
chat.endpoint(method: "GET", path: "/", handler: "index")
chat.endpoint(method: "GET", path: "/ws", handler: "ws_handler")
chat.endpoint(method: "GET", path: "/stats", handler: "stats")
server.service(chat)
server.compile()

File: examples-sdk/kotlin/010-basic-websocket-chat/main.kt
// 010-basic-websocket-chat — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 010-basic-websocket-chat/main.kt --release

fun main() {
    val server = VilServer("websocket-chat", 8080)
    val chat = ServiceProcess("chat")
    chat.endpoint("GET", "/", "index")
    chat.endpoint("GET", "/ws", "ws_handler")
    chat.endpoint("GET", "/stats", "stats")
    server.service(chat)
    server.compile()
}
